lib/src/facts/linux/os_linux.cc

```diff
@@ -265,6 +265,7 @@ namespace facter { namespace facts { namespace linux {
                 { string(os::oracle_enterprise_linux),  string(release_file::oracle_enterprise_linux) },
                 { string(os::oracle_vm_linux),          string(release_file::oracle_vm_linux) },
                 { string(os::arista_eos),               string(release_file::arista_eos) },
+                { string(os::gentoo),                   string(release_file::gentoo) },
```

Comment:
Looks like this works because the 1st thing we do after is look for the pattern `release (\d[\d.]*)` in the first line. That matches `Gentoo Base System release 2.2`, and the rest falls together.